### ISSUE TITLE
`convert` modifies name fields from parsed entries

### ISSUE THREAD

**Issue by RubenVerborgh** [inukshuk/bibtex-ruby#118]
The `convert` method, as opposed to `convert!`, is not supposed to modify an entry. Even though there is a [test](https://github.com/inukshuk/bibtex-ruby/blob/b13747a4e5663e19256828c134534fe77c3bea2e/test/bibtex/test_entry.rb#L413-L417) for this, it fails when fields of the entry are set through parsing instead of directly:

```ruby
entry = BibTeX.parse('@book{o, author = {M\\"{o}by Dick} }')[0]
# entry.author = entry.author.to_s # workaround for the bug

puts entry[:author] # writes M\\"{o}by Dick
entry.convert :latex
puts entry[:author] # writes Möby Dick, but expected to be M\\"{o}by Dick
```

Below is a suggested unit test to place after the current test (but a fix went too deep into the internals for me):

```ruby
it 'does not change the author of a parsed entry' do
  entry = BibTeX.parse('@book{o, author = {M\\"{o}by Dick}}')[0]
  e = entry.convert(:latex)
  e.wont_be :==, entry
  e.author.to_s.length.must_be :<, entry.author.to_s.length
end

```

**Comment by inukshuk** [comment]
Aw, thanks, that was pretty big -- name variables were not duplicated correctly (so the actual string used for the name parts stayed the same). The existing test worked because the title variable was a string value (where duplication worked). Good find!

Should be fixed in 4.4.1

**Comment by RubenVerborgh** [comment]
Thanks a lot!
